Add some rate limiting information for GOV.UK Notify (#2639)
alphagov/govuk-developer-docs: source/manual/govuk-notify.html.md

  email updates about pieces of content on GOV.UK. It's our biggest sender of
  emails and regularly exceeds one million emails per day.

  **Note:** currently (as of 30/06/2020) GOV.UK Notify has a maximum rate limit of 350
  requests per second meaning we should adhere to this limit when using Notify to

>   deliver emails for our applications. An example of where we set this for

Reviewer: ```suggestion
  deliver emails for our applications. This is set in...
```

We only implement the limit in one place, since we have a single/unified method of sending email.

>   EmailAlertApi can be found in the [DeliveryRequestWorker][DeliveryRequestWorker].

Reviewer: ```suggestion
  email-alert-api can be found in the [DeliveryRequestWorker][DeliveryRequestWorker].
```

  example might be to receive an email when a scheduled document on GOV.UK has
  been automatically published.

> Currently (as of 30/06/2020) GOV.UK Notify has a maximum rate limit of 350

Reviewer: This only applies to "GOV.UK Emails". Can we move it as a note under that ☝️ ?

Author: Sure

requests per second meaning we should adhere to this limit when using Notify to
deliver emails for our applications. An example of where we set this for
EmailAlertApi can be found in the [DeliveryRequestWorker][DeliveryRequestWorker].

If you wish to double check this figure you could ask in their slack channel #govuk-notify.

[DeliveryRequestWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/delivery_request_worker.rb#L52

> _In the future we may set up a new 'GOV.UK Public' service which is used to

Reviewer: Out-of-scope, but can we remove this line? It's really strange to see a "idea" in our developer docs.
